lib/manager/components/LabelEditor.js

```diff
@@ -0,0 +1,234 @@
+      labelIsNew: false
+    }
+
+    componentDidMount = () => {
+      // If we didn't get a label passed in, declare the label to be new
+      if (Object.keys(this.props.label).length === 0) {
+        this.setState({
+          labelIsNew: true
+        })
+      }
+    }
+
+    userIsAdmin = (user: ManagerUserState) => {
+      const {permissions} = user
+      if (!permissions) return false
+      else return permissions.isApplicationAdmin() || permissions.canAdministerAnOrganization()
+    }
+
+  _onFormChange = ({target}: {target: HTMLInputElement}) => {
+    const {name, value} = target
+
+    this.setState(
+      update(this.state, {
+        newLabel: { $merge: {[name]: value} },
+        validation: { [name]: { $set: value && value.length > 0 } }
+      })
+    )
+  }
+
+  _onCheckboxChange = ({target}: {target: HTMLInputElement}) => {
+    const {name, checked} = target
+
+    this.setState(
+      update(this.state, {
+        newLabel: { $merge: {[name]: checked} },
+        // Checkboxes are valid in any state
+        validation: { [name]: { $set: true } }
+      })
+    )
+  }
+
+  /**
+   * Compares the dirty label from the old one to determine which keys changed
+   * @returns Object with all changes
+   */
+  _getChanges = () => {
+    const newLabel = this.state.newLabel
+    const oldLabel = this.props.label
+
+    const changes: any = {}
+    Object.keys(newLabel).forEach(k => {
+      if (newLabel[k] !== oldLabel[k]) {
+        changes[k] = newLabel[k]
+      }
+    })
+    return changes
+  }
+
+  _settingsAreUnedited = () => Object.keys(this._getChanges()).length === 0
+
+  _formIsValid = () => {
+    const {validation} = this.state
+    return Object.keys(validation).every(k => validation[k])
+  }
+
+  _onSaveSettings = () => {
+    const {labelIsNew, newLabel: label} = this.state
+    const {createLabel, updateLabel} = this.props
+
+    if (labelIsNew) {
+      // Insert project ID
+      label.projectId = this.props.projectId
+      createLabel(label)
+    } else {
+      updateLabel(label)
+    }
+
+    // This passed method will close the editor
+    this.props.onDone && this.props.onDone()
+  }
+
+  render () {
+    const { newLabel, validation } = this.state
+    const { user } = this.props
+
+    return (
+      <Form>
+        <Panel>
+          <ListGroup fill>
+            <ListGroupItem>
+              <FormGroup validationState={validationState(validation.name)}>
+                <ControlLabel>Name</ControlLabel>
+                <FormControl
+                  value={newLabel.name}
+                  name={'name'}
+                  onChange={this._onFormChange}
+                />
+                <FormControl.Feedback />
+                <HelpBlock>Required.</HelpBlock>
+              </FormGroup>
+              <FormGroup>
+                <ControlLabel>Description</ControlLabel>
+                <FormControl
+                  value={newLabel.description}
+                  name={'description'}
```

Comment:
When creating a new label, if you change the description the modal will "let" you save the label without entering a name (that's caught server-side).

![Screen Recording 2021-07-30 at 12 25 05 PM](https://user-images.githubusercontent.com/56846598/127683383-de0b1e29-acf8-478d-9753-a546482afcf8.gif)

Reply:
Really good catch thanks! Have refactored the validation a bunch